external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaOffsetReaderSuite.scala

```diff
@@ -66,74 +70,124 @@ class KafkaOffsetReaderSuite extends QueryTest with SharedSparkSession with Kafk
     )
   }
 
+  test("isolationLevel must give back default isolation level when not set") {
+    testIsolationLevel(None,
+      IsolationLevel.valueOf(ConsumerConfig.DEFAULT_ISOLATION_LEVEL.toUpperCase(Locale.ROOT)))
+  }
+
+  test("isolationLevel must give back READ_UNCOMMITTED when set") {
+    testIsolationLevel(Some("read_uncommitted"), IsolationLevel.READ_UNCOMMITTED)
+  }
+
+  test("isolationLevel must give back READ_COMMITTED when set") {
+    testIsolationLevel(Some("read_committed"), IsolationLevel.READ_COMMITTED)
+  }
+
+  test("isolationLevel must throw exception when invalid isolation level set") {
+    intercept[IllegalArgumentException] {
+      testIsolationLevel(Some("intentionally_invalid"), IsolationLevel.READ_COMMITTED)
+    }
+  }
+
+  private def testIsolationLevel(kafkaParam: Option[String], isolationLevel: IsolationLevel) = {
+    var kafkaParams = Map(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG -> testUtils.brokerAddress)
+    kafkaParam.foreach(p => kafkaParams ++= Map(ConsumerConfig.ISOLATION_LEVEL_CONFIG -> p))
+    val reader = new KafkaOffsetReaderAdmin(
+      SubscribeStrategy(Seq()),
+      KafkaSourceProvider.kafkaParamsForDriver(kafkaParams),
+      CaseInsensitiveMap(Map.empty),
+      ""
+    )
+    assert(reader.isolationLevel === isolationLevel)
+  }
+
   test("SPARK-30656: getOffsetRangesFromUnresolvedOffsets - using specific offsets") {
```

Comment:
I'd rather follow the way to create tests dynamically.

https://github.com/apache/spark/blob/e43255051c0a82713d653fe590fe7728e43556ce/sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingAggregationSuite.scala#L53-L103

This has a downside to not able to run test individually in IDE, but at least this will ensure each test runs individually. Adding `Seq` in test you don't know test fails with which option, "true" or "false".

Reply:
OK, changing. That makes more clear which test fails.